R/plots.R

```diff
@@ -125,6 +125,187 @@ plot_pca <- function(obj,
   p
 }
 
+
+#' Plot Loadings and Interpretations 
+#' 
+#' give a principal component, tells you which contribute the most or give a sample, tells you which PC's it contributes to the most
+#' 
+#' @param obj a \code{sleuth} object
+#' @param use_filtered if TRUE, use filtered data. otherwise, use all data
+#' @param sample user input on which sample and which PC's contribute the most
+#' @param PC  principal component to view sample's contribution to that PC
+#' @param units either 'est_counts' or 'tpm'
+#' @param pc_count # of PC's
+#' @param bool scale or not
+#' @param absolute default true, to see all PC's magnitude (recommended)
+#' @return a ggplot object
+#' @export
+plot_loadings <- function(obj, 
+  use_filtered = TRUE,
+  sample = NULL, #string please
+  PC = NULL, #apparently if u type in a string or an integer (corresponding to the PC), they're both ok
+  units = 'est_counts',
+  pc_count = NULL,
+  bool = FALSE,
+  absolute = TRUE, 
+  ...) {
+
+  stopifnot( is(obj, 'sleuth') )
+  #filtering?? doesn't work right now
+
+  # mat <- NULL
+  # if (use_filtered) {
+  #   mat <- spread_abundance_by(obj$obs_norm_filt, units)
+  # } else {
+  #   mat <- spread_abundance_by(obj$obs_norm, units)
+  # }
+  mat <- spread_abundance_by(obj$obs_norm, units)
+  
+  pca_calc <- prcomp(mat, scale = bool)
+
+  #sort of hack-y, may wish to fix
+  if (sample == '') {
+    sample <- NULL
+  }
+
+  loadings <- t(pca_calc$x)
+
+  #debugging
+  # print(sample)
+  # print(summary(loadings))
+  # print(pc_count)
+  # print("pc")
+  # print(PC)
+
+
+  executed <- FALSE
+  #given a sample
+  if (!is.null(sample)) {
+    executed <- TRUE
+    loadings <- pca_calc$x[sample,]
+    if (absolute) {
+      loadings <- abs(loadings)
+    }
+    loadings <- sort(loadings, decreasing = TRUE)
+    names <- names(loadings)
+  }
+
+  #given a PC, which samples contribute the most?
+  if (!executed) {
+    loadings <- pca_calc$x[,PC]
+    #print('reached here')
+    #print(loadings)
+    if (absolute) {
+      loadings <- abs(loadings)
+    }
+    loadings <- sort(loadings, decreasing = TRUE)
+    names <- names(loadings)
```

Comment:
by doing this, you are overwriting the function `names` anywhere after in this function. if someone touches us later and actually needs to use the function `names`, it might lead to some strange bugs. please try not to override common function names. another 1 of these is `var`